pkg/kubectl/cmd/get/get_test.go

```diff
@@ -1324,34 +1446,63 @@ func TestWatchResourceIdentifiedByFile(t *testing.T) {
 		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
 		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
 			switch req.URL.Path {
-			case "/namespaces/test/replicationcontrollers/cassandra":
+			case "/namespaces/test/pods/busybox1":
 				return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: cmdtesting.ObjBody(codec, &pods[1])}, nil
-			case "/namespaces/test/replicationcontrollers":
-				if req.URL.Query().Get("watch") == "true" && req.URL.Query().Get("fieldSelector") == "metadata.name=cassandra" {
-					return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: watchBody(codec, events[1:])}, nil
-				}
-				t.Fatalf("request url: %#v,and request: %#v", req.URL, req)
-				return nil, nil
 			default:
-				t.Fatalf("request url: %#v,and request: %#v", req.URL, req)
-				return nil, nil
+				t.Errorf("request url: %#v,and request: %#v", req.URL, req)
+				return nil, errors.New("internal error")
 			}
 		}),
 	}
 
-	streams, _, buf, _ := genericclioptions.NewTestIOStreams()
+	tf.FakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme.Scheme, toUnstructured(podsToObjects(pods[1])...)...)
+	tf.FakeDynamicClient.PrependReactor("list", "*", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		fieldSelector := action.(clienttesting.ListAction).GetListRestrictions().Fields.String()
+		if fieldSelector != "metadata.name=busybox1" {
+			t.Errorf("bad field selector: %q", fieldSelector)
+		}
+		return false, nil, nil
+	})
+	tf.FakeDynamicClient.PrependWatchReactor("*", func(action clienttesting.Action) (handled bool, ret watch.Interface, err error) {
+		fieldSelector := action.(clienttesting.WatchAction).GetWatchRestrictions().Fields.String()
+		if fieldSelector != "metadata.name=busybox1" {
+			t.Errorf("bad field selector: %q", fieldSelector)
+		}
+
+		evs := events[2:]
+		ch := make(chan watch.Event, len(evs))
+		for _, e := range evs {
+			ch <- e
+		}
+		return true, watch.NewProxyWatcher(ch), nil
+	})
+
+	buf := &cmdtesting.ThreadsafeBuffer{}
+	streams := genericclioptions.IOStreams{
+		In:     &bytes.Buffer{},
+		Out:    buf,
+		ErrOut: &bytes.Buffer{},
+	}
 	cmd := NewCmdGet("kubectl", tf, streams)
 	cmd.SetOutput(buf)
 
 	cmd.Flags().Set("watch", "true")
-	cmd.Flags().Set("filename", "../../../../test/e2e/testing-manifests/statefulset/cassandra/controller.yaml")
```

Comment:
Why you're changing test contents? I'd like to see tests verifying different types of objects, not just pods.

Reply:
Returning `Pods` for `.../namespaces/test/replicationcontrollers` is a) wrong b) breaks the proper fakes actually checking GVK

Reply:
It took me like and hour to fix the actual watch code and 2 days fixing the unit (and other) tests because of these things